merlin/analysis/optimize.py

```diff
@@ -59,9 +65,25 @@ def _run_analysis(self, fragmentIndex):
                 self.parameters['preprocess_task'])
         codebook = self.get_codebook()
 
-        fovIndex = np.random.choice(list(self.dataSet.get_fovs()))
-        zIndex = np.random.choice(
-            list(range(len(self.dataSet.get_z_positions()))))
+        # Use random FOV and Z index if the user did not specify.
+        if 'fov_index' not in self.parameters:
+            fovIndex = np.random.choice(list(self.dataSet.get_fovs()))
+            zIndex = np.random.choice(
+                list(range(len(self.dataSet.get_z_positions()))))
```

Comment:
Can we move this to the constructor? Then when fovs are randomly selected, they will be saved into the parameters for the analysis task to better record which fovs the optimization was run on.